Juego.cpp

```diff
@@ -0,0 +1,278 @@
+        }
+
+        jugador = new Jugador(ficha, cantidadDeFichas);
+        this->jugadores->agregarFin(jugador);
+    }
+
+}
+
+
+int Juego::getCantidadFichasEnLinea(){
+    
+    return this->cantidadFichasEnLinea;
+
+}
+
+
+//PUBLIC
+Juego::Juego(int cantidadFilas, int cantidadColumnas, int profundidad, int cantidadJugadores, int cantidadCartasDelMazo, int cantidadDeFichas, int fichasEnLinea){
+
+        this->jugadores = new Lista<Jugador*>;
+        this->mazoJuego = new Mazo(cantidadCartasDelMazo);
+        this->tablero = new Tablero(cantidadColumnas, cantidadFilas, profundidad);
+        this->numeroJugadorActual = 0;
+        this->cantidadFichasEnLinea = fichasEnLinea;
+        this->estadoActual = Jugando;
+        this->cartaActualElegida = Ninguna;
+
+        this->cargarJugadores(cantidadJugadores);
+        
+}
+
+
+int Juego::getNumeroJugadorActual(){
+
+    return this->numeroJugadorActual;
+
+}
+
+
+void Juego::setNumeroJugadorActual(int numeroJugador){
+
+    if(numeroJugador < 0 && numeroJugador <= this->jugadores->len() ){
+        throw("El numero del jugador no existe");
+    }
+    this->numeroJugadorActual = numeroJugador;
+
+}
+
+
+bool Juego::estaBloqueado(){
+
+    return (this->jugadores->obtener(this->getNumeroJugadorActual())->getTipoDeTurno() == Bloqueado); 
+
+}
+
+
+void Juego::agarrarCarta(){
+    
+    if( this->tieneMenosDeTresCartas() ){
+        this->jugadores->obtener(this->getNumeroJugadorActual())->cargarCarta(this->mazoJuego->darCarta());
+    }
+
+}
+
+
+bool Juego::tieneLaCartaElegida(CartaElegida cartaSeleccionada){
+
+    return ( this->jugadores->obtener(this->getNumeroJugadorActual())->getCantidadDeCartasEnMano >= cartaSeleccionada );
+
+}
+
+
+bool Juego::arrojoCarta(){
+
+    return ( this->getCartaElegida() != Ninguna );
+
+}
+
+
+void Juego::setCartaElegida(CartaElegida cartaSeleccionada){
+
+    this->cartaActualElegida = cartaSeleccionada;
+
+}
+
+
+CartaElegida Juego::getCartaElegida(){
+
+    return this->cartaActualElegida; 
+
+}
+
+
+void Juego::usarCarta(CartaElegida cartaSeleccionada){
+  
+    switch(this->jugadores->obtener(this->getNumeroJugadorActual())->obtenerCarta(cartaSeleccionada)->getCartaEspecial()){
+        case JuegaDoble: 
+            this->jugadores->obtener(this->getNumeroJugadorActual())->duplicarTurno();
+        break;
+        case BloquearTurno:
+            this->jugadores->obtener(this->getNumeroSiguienteJugador())->bloquearTurno();
+        break;
+        case AgarrarCincoFichas:
+            this->jugadores->obtener(this->getNumeroJugadorActual())->sumarCincoFichas();
+        break;
+        case EliminarMazoSiguiente:
+            while (this->jugadores->obtener(this->getNumeroSiguienteJugador())->getCantidadDeCartasEnMano() > 0){
+                this->jugadores->obtener(this->getNumeroSiguienteJugador())->eliminarCarta(this->jugadores->obtener(this->getNumeroSiguienteJugador())->getCantidadDeCartasEnMano());
```

Comment:
Acá me tira un error raro: `argument of type "int" is incompatible with parameter of type "CartaElegida"`.

Parece que `eliminarCarta()` está esperando que le pases `Ninguna`, `Primera`, `Segunda` o `Tercera`; que son los valores del enum `CartaElegida`. Sin embargo, lo que hay dentro de los paréntesis devuelve `0`, `1`, `2` o `3`, los cuales al parecer no detecta como lo mismo.

Propongo que cambies que `eliminarCarta()` acepte datos del tipo `<int >` para esa función, o si es incompatible con otras funciones, la otra es simplemente hacer una función igual, pero que funciones con int:
```c++
void Jugador::eliminarCarta(int posicionCarta){

    this->cartasJugador->pop(posicionCarta - 1);

}
```